packages/utilities/src/CliUtils.ts

```diff
@@ -534,6 +535,35 @@ export class CliUtils {
         return answerToReturn;
     }
 
+    public static async interactiveSelection(menu: string[], opts?: IInteractiveOptions): Promise<number> {
+        return new Promise<number>(async (resolve, reject) => {
+            // TODO(Fernando) - Move terminal code out of the API
+            // TODO(Fernando) - Add corresponding `@types/terminal-kit`
+            const term = require("terminal-kit").terminal;
+            if (opts?.header != null) term.defaultColor(opts.header);
+            try {
+                term.grabInput(true);
+                const cursor = await term.getCursorLocation();
+                term.on("key", (name: string) => {
+                    if (name === "CTRL_C" || name === "ESCAPE") {
+                        if (name === "CTRL_C") {
+                            term.moveTo(1, cursor.y + menu.length + (opts?.header != null ? 1 : 0) + 1);
+                        }
+                        term.grabInput(false);
+                        reject("No input selected!");
+                    }
+                });
+                const singleColumnMenu = require("util").promisify(term.singleColumnMenu);
+                const userInteraction = await singleColumnMenu.call(term, menu, {cancelable: true});
+                term.grabInput(false);
+                resolve(userInteraction.selectedIndex + 1);
+            } catch(err) {
+                term.grabInput(false);
+                reject("Oh oh! Something went wrong!\nDetails:\n" + err.message);
```

Comment:
"Oh oh" -> "Uh oh"

Also same question as above, should an ImperativeError be generated here?